### PR TITLE
refactor: Migrate EventMapper to Typescript

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

**Review by ffflorian — CHANGES_REQUESTED** [review]
Please remove all JSDoc typings (`@param {Conversation} conversationEntity ...` => `@param conversationEntity ...`).

**Comment by thisisamir98** [comment]
@ffflorian done in https://github.com/wireapp/wire-webapp/pull/9647/commits/8d63b0017dc410702375c44647ec809b3ef6fe2e

File: src/script/conversation/EventMapper.ts [hunk 1]
```diff
@@ -404,10 +414,10 @@ export class EventMapper {
    * Map JSON ata of group creation event to message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {MemberMessage} Member message entity
+   * @param eventData Message data
+   * @returns Member message entity
    */
-  _mapEventGroupCreation({data: eventData}) {
+  _mapEventGroupCreation({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventGroupCreation({data: eventData}: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 2]
```diff
@@ -354,10 +364,10 @@ export class EventMapper {
    * Maps JSON data of conversation.one2one-creation message into message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {ContentMessage} Member message entity
+   * @param eventData Message data
+   * @returns Member message entity
    */
-  _mapEvent1to1Creation({data: eventData}) {
+  _mapEvent1to1Creation({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEvent1to1Creation({data: eventData}: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 3]
```diff
@@ -375,10 +385,10 @@ export class EventMapper {
    * Maps JSON data of conversation.asset_add message into message entity.
    *
    * @private
-   * @param {Object} event Message data
-   * @returns {ContentMessage} Content message entity
+   * @param event Message data
+   * @returns Content message entity
    */
-  _mapEventAssetAdd(event) {
+  _mapEventAssetAdd(event: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventAssetAdd(event: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 4]
```diff
@@ -391,10 +401,10 @@ export class EventMapper {
    * Maps JSON data of delete everywhere event to message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {DeleteMessage} Delete message entity
+   * @param eventData Message data
+   * @returns Delete message entity
    */
-  _mapEventDeleteEverywhere({data: eventData}) {
+  _mapEventDeleteEverywhere({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventDeleteEverywhere({data: eventData}: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 5]
```diff
@@ -416,18 +426,18 @@ export class EventMapper {
...
    * Maps JSON data of conversation.location message into message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {ContentMessage} Location message entity
+   * @param eventData Message data
+   * @returns Location message entity
    */
-  _mapEventLocation({data: eventData}) {
+  _mapEventLocation({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventLocation({data: eventData}: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 6]
```diff
@@ -446,11 +456,11 @@ export class EventMapper {
...
    * @private
-   * @param {Object} event Message data
-   * @param {Conversation} conversationEntity Conversation entity the event belong to
-   * @returns {MemberMessage} Member message entity
+   * @param event Message data
+   * @param conversationEntity Conversation entity the event belong to
+   * @returns Member message entity
    */
-  _mapEventMemberJoin(event, conversationEntity) {
+  _mapEventMemberJoin(event: EventRecord, conversationEntity: Conversation) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventMemberJoin(event: EventRecord, conversationEntity: Conversation) {
```

File: src/script/conversation/EventMapper.ts [hunk 7]
```diff
@@ -483,10 +493,10 @@ export class EventMapper {
    * Maps JSON data of conversation.member_leave message into message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {MemberMessage} Member message entity
+   * @param eventData Message data
+   * @returns Member message entity
    */
-  _mapEventMemberLeave({data: eventData}) {
+  _mapEventMemberLeave({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private _mapEventMemberLeave({data: eventData}: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 8]
```diff
@@ -496,10 +506,10 @@ export class EventMapper {
    * Maps JSON data of conversation.message_add message into message entity.
    *
    * @private
-   * @param {Object} event Message data
-   * @returns {Promise<ContentMessage>} Content message entity
+   * @param event Message data
+   * @returns Content message entity
    */
-  async _mapEventMessageAdd(event) {
+  async _mapEventMessageAdd(event: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private async _mapEventMessageAdd(event: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 9]
```diff
@@ -516,18 +526,20 @@ export class EventMapper {
     return messageEntity;
   }
 
-  async _mapEventCompositeMessageAdd(event) {
+  async _mapEventCompositeMessageAdd(event: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]:
```suggestion
  private async _mapEventCompositeMessageAdd(event: EventRecord) {
```

File: src/script/conversation/EventMapper.ts [hunk 10]
```diff
@@ -555,10 +567,10 @@ export class EventMapper {
    * Maps JSON data of conversation.rename message into message entity.
    *
    * @private
-   * @param {Object} eventData Message data
-   * @returns {RenameMessage} Rename message entity
+   * @param eventData Message data
+   * @returns Rename message entity
    */
-  _mapEventRename({data: eventData}) {
+  _mapEventRename({data: eventData}: EventRecord) {
```
Comment (ffflorian) [inline src/script/conversation/EventMapper.ts]: etc...

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: src/script/conversation/EventMapper.ts (hunk 10 above)
Comment (thisisamir98) [reply src/script/conversation/EventMapper.ts]: ?

**Review by ffflorian — CHANGES_REQUESTED** [review]
Please also mark the other functions which are tagged with `@private` as private in TypeScript. You can remove the `@private` tag.

**Comment by thisisamir98** [comment]
@ffflorian private methods in https://github.com/wireapp/wire-webapp/pull/9647/commits/7fa9ece49873e1e357dd5c89d241e767030f903a

**Review by ffflorian — APPROVED** [review]

**Pull request merged**